modules/ROOT/pages/deployment/services/s-list/postprocessing.adoc

```diff
@@ -34,15 +34,29 @@ When postprocessing has been enabled, configuring any postprocessing step will r
+To enable virus scanning as a postprocessing step after uploading a file, the environment variable `POSTPROCESSING_STEPS` needs to contain the word `virusscan` at one location in the list of steps. As a result, each uploaded file gets virus scanned as part of the postprocessing steps. Note that the `antivirus service` is required to be enabled and configured for this to work.
 
 === Delay
 
-Though this is for development purposes only and NOT RECOMMENDED on production systems, setting the environment variable `POSTPROCESSING_DELAY` to a duration not equal to zero will add a delay step with the configured amount of time. ocis will continue postprocessing the file after the configured delay.
+Though this is for development purposes only and NOT RECOMMENDED on production systems, setting the environment variable `POSTPROCESSING_DELAY` to a duration not equal to zero will add a delay step with the configured amount of time. Infinite Scale will continue postprocessing the file after the configured delay. Use the enviroment variable `POSTPROCESSING_STEPS` and the keyword `delay` if you have multiple postprocessing steps and want to define their order. If `POSTPROCESSING_DELAY` is set but the keyword `delay` is not contained in `POSTPROCESSING_STEPS`, it will be processed as last postprocessing step without being listed there. In this case, a log entry will be written on service startup to notify the admin about that situation. That log entry can be avoided by adding the keyword `delay` to `POSTPROCESSING_STEPS`.
+
+== Custom Postprocessing Steps
+
+By using the envvar `POSTPROCESSING_STEPS`, custom postprocessing steps can be added. Any word can be used as step name but be careful not to conflict with exising keywords like `virusscan` and `delay`. In addition, if a keyword is misspelled or the corresponding service does either not exist or does not follow the necessary event communication, the postprocessing service will wait forever getting the required response to proceed and does not continue any other processing.
+
+=== Prerequisites
+
+For using custom postprocessing steps, you need a custom service listening to the configured event system, see xref:general-prerequisites[General Prerequisites]
+
+=== Workflow
+
+When setting a custom postprocessing step (eg. "customstep") the postprocessing service will eventually sent an event during postprocessing. The event will be of type StartPostprocessingStep with its field StepToStart set to "customstep". When the custom service receives this event, it can savely execute its actions. The postprocessing service will wait until it has finished its work. The event contains further information (like filename, executing user, size, ...) and required tokens respectively urls to download the file in case byte inspection is necessary.
```

Comment:
```suggestion
When setting a custom post-processing step (eg. "customstep") the post-processing service will eventually send an event during post-processing. The event will be of the type StartPostprocessingStep with its field StepToStart set to "customstep". When the custom service receives this event, it can safely execute its actions. The post-processing service will wait until it has finished its work. The event contains further information (like filename, executing user, size, ...) and required tokens respectively URLs to download the file in case byte inspection is necessary.
```